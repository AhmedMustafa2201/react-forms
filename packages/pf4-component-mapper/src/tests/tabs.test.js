import React from 'react';
import toJson from 'enzyme-to-json';
import Tabs from '../form-fields/tabs';
import { mount, shallow } from 'enzyme';

describe('Tabs component', () => {
  const props = {
    fields: [
      {
        title: 'cosiTitle',
        name: 'cosiName',
        fields: [],
      },
      {
        title: 'cosiTitle2',
        name: 'cosiName2',
        fields: [],
      },
    ],
    formOptions: {
      renderForm: (fields, formOptions) => <div>{ 'Here would be form' }</div>,
    },
  };

  it('should render tabs correctly', () => {
    const wrapper = shallow(
      <Tabs { ...props }>
      </Tabs>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should switch tabs correctly', () => {
    const wrapper = mount(<Tabs { ...props } />);
    expect(wrapper.instance().state.activeTabKey).toEqual(0);

    const secondTabButton = wrapper.find('.pf-c-tabs__button').at(6);
    secondTabButton.simulate('click');

    expect(wrapper.instance().state.activeTabKey).toEqual(1);
  });
});
