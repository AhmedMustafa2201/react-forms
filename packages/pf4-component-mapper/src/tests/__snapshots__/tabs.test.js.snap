// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component should render tabs correctly 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "restart": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <FormTabs
      fields={
        Array [
          Object {
            "fields": Array [],
            "name": "cosiName",
            "title": "cosiTitle",
          },
          Object {
            "fields": Array [],
            "name": "cosiName2",
            "title": "cosiTitle2",
          },
        ]
      }
    />
  </ContextProvider>
</ContextProvider>
`;
